backend/danswer/connectors/teams/connector.py

```diff
@@ -0,0 +1,269 @@
+import os
+from datetime import datetime
+from html.parser import HTMLParser
+from typing import Any
+
+import msal  # type: ignore
+from office365.graph_client import GraphClient  # type: ignore
+from office365.teams.channels.channel import Channel  # type: ignore
+from office365.teams.chats.messages.message import ChatMessage  # type: ignore
+from office365.teams.team import Team  # type: ignore
+
+from danswer.configs.app_configs import INDEX_BATCH_SIZE
+from danswer.configs.constants import DocumentSource
+from danswer.connectors.interfaces import GenerateDocumentsOutput
+from danswer.connectors.interfaces import LoadConnector
+from danswer.connectors.interfaces import PollConnector
+from danswer.connectors.interfaces import SecondsSinceUnixEpoch
+from danswer.connectors.models import BasicExpertInfo
+from danswer.connectors.models import ConnectorMissingCredentialError
+from danswer.connectors.models import Document
+from danswer.connectors.models import Section
+from danswer.utils.logger import setup_logger
+
+# import pptx  # type: ignore
+
+logger = setup_logger()
+
+
+class HTMLFilter(HTMLParser):
+    text = ""
+
+    def handle_data(self, data: str) -> None:
+        self.text += data
+
+
+def get_created_datetime(obj: ChatMessage) -> datetime:
+    # Extract the 'createdDateTime' value from the 'properties' dictionary
+    created_datetime_str = obj.properties["createdDateTime"]
+
+    # Convert the string to a datetime object
+    return datetime.strptime(created_datetime_str, "%Y-%m-%dT%H:%M:%S.%f%z")
```

Comment:
there's also a function for this. instead of creating get_created_datetime you can just call the existing function with obj.properties["createdDateTime"]